rollout/sync.go

```diff
@@ -561,13 +561,27 @@ func (c *rolloutContext) calculateRolloutConditions(newStatus v1alpha1.RolloutSt
 
 	if isAborted {
 		revision, _ := replicasetutil.Revision(c.rollout)
+		var analysisRunStatusCanaryStep, analysisRunStatusCanaryBackground, analysisRunStatus v1alpha1.AnalysisPhase
```

Comment:
is there a rollout status that provides analysis failed or is this the only way to find the analysis status